src/main/java/net/imglib2/img/ImgView.java

```diff
@@ -154,25 +156,47 @@ public Object subIntervalIterationOrder( final Interval interval )
 		if ( this.sourceInterval instanceof SubIntervalIterable )
 			return ( ( SubIntervalIterable< T > ) this.sourceInterval ).localizingCursor( interval );
 		else
-			return Views.interval( this.sourceInterval, interval )
-					.localizingCursor();
+			return Views.interval( this.sourceInterval, interval ).localizingCursor();
 	}
 
 	/**
-	 * Represent an arbitrary RandomAccessibleInterval as an Img
+	 * Represent an arbitrary {@link RandomAccessibleInterval} as an
+	 * {@link Img}, with a suitable {@link ImgFactory} for its size and type,
+	 * created by
+	 * {@link Util#getSuitableImgFactory(net.imglib2.Dimensions, Object)}.
 	 *
 	 * @param accessible
-	 * 			RandomAccessibleInterval which will be wrapped with an ImgView
-	 * @param factory
-	 * 			ImgFactory returned by {@link ImgView#factory()}
-	 * @return
-	 * 			RandomAccessibleInterval represented as an Img
+	 *            RandomAccessibleInterval which will be wrapped with an ImgView
+	 * @return RandomAccessibleInterval represented as an Img
+	 * @see Util#getSuitableImgFactory(net.imglib2.Dimensions, Object)
 	 */
-	public static < T extends Type< T >> Img< T > wrap(final RandomAccessibleInterval< T > accessible, final ImgFactory< T > factory){
-		if(accessible instanceof Img){
-			return (Img<T>) accessible;
-		}else{
-			return new ImgView< T >(accessible, factory);
+	public static < T extends Type< T > > Img< T > wrap( final RandomAccessibleInterval< T > accessible )
```

Comment:
In order to avoid a breaking change, should we keep the existing singature for `wrap`? Maybe deprecate?

Reply:
I did. And deprecated it. No?

Reply:
Oh yes, you did. I just didn't look carefully enough, my bad.